### PR TITLE
Add last post link and profile links to subforum topic entries

### PR DESCRIPTION
**Pull request by Noxxaa**
Fixes regressions compared to the original forum. Profile links help finding user information, and latest post link icon helps with forum browsing in general.

Old:

image

New:

image

Old forum for comparison:

image

### CODE REVIEW

**Comment by Noxxaa** [comment]
Addresses issue #407

File: TASVideos/Pages/Forum/Subforum/Index.cshtml [hunk 1]
```diff
@@ -38,8 +38,15 @@
...
 				</td>
 				<td>@Math.Max(0, topic.PostCount - 1)</td> @*Imported Submission Topics will have no replies and no original post*@
-				<td>@topic.CreateUserName</td>
-				<td><timezone-convert asp-for="@topic.LastPostDateTime" /></td>
+				<td> <profile-link username="@topic.CreateUserName">@topic.CreateUserName</profile-link></td>
+                <td>
+                    @if (topic.LastPost != null)
+                    {
+                        <timezone-convert asp-for="@topic.LastPostDateTime" /> <br />
+                        <profile-link username="@topic.LastPostUserName">@topic.LastPostUserName</profile-link>
```
Comment (adelikat) [inline TASVideos/Pages/Forum/Subforum/Index.cshtml:46]: This taghelper allows you to omit the value between the tags.  so you can simply do:
<profile-link username="@topic.LastPostUserName"></profile-link>

File: TASVideos/Pages/Forum/Subforum/Index.cshtml.cs [hunk 2]
```diff
@@ -62,7 +62,7 @@ public async Task<IActionResult> OnGet()
 					CreateTimestamp = ft.CreateTimestamp,
 					Type = ft.Type,
 					PostCount = ft.ForumPosts.Count,
-					LastPost = ft.ForumPosts.Max(fp => (DateTime?)fp.CreateTimestamp)
+					LastPost = ft.ForumPosts.OrderBy(fp => fp.CreateTimestamp).Last()
```
Comment (adelikat) [inline TASVideos/Pages/Forum/Subforum/Index.cshtml.cs:65]: I wonder if .OrderByDescending().First() is better.  EF might be smart enough to do the same query either way, but it is worth checking

Comment (Noxxaa) [reply TASVideos/Pages/Forum/Subforum/Index.cshtml.cs]: Based on my understanding it is guaranteed at least not to be worse, so I've made the change in the next commit

**Pull request merged**